pkg/util/v1alpha2/metricscollector/metricscollector.go

```diff
@@ -0,0 +1,116 @@
+package metricscollector
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+
+	v1alpha2 "github.com/kubeflow/katib/pkg/api/v1alpha2"
+)
+
+type MetricsCollector struct {
+	clientset *kubernetes.Clientset
+}
+
+func NewMetricsCollector() (*MetricsCollector, error) {
+	config, err := restclient.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &MetricsCollector{
+		clientset: clientset,
+	}, nil
+
+}
+
+func (d *MetricsCollector) CollectObservationLog(tId string, jobKind string, objectiveValueName string, metrics []string, namespace string) (*v1alpha2.ObservationLog, error) {
+	labelMap := make(map[string]string)
+
+	// TODO: Create labelMap based on jobKind
+
+	pl, _ := d.clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: labels.Set(labelMap).String(), IncludeUninitialized: true})
+	if len(pl.Items) == 0 {
+		return nil, errors.New(fmt.Sprintf("No Pods are found in Trial %v", tId))
+	}
+	logopt := apiv1.PodLogOptions{Timestamps: true}
+	logs, err := d.clientset.CoreV1().Pods(namespace).GetLogs(pl.Items[0].ObjectMeta.Name, &logopt).Do().Raw()
+	if err != nil {
+		return nil, err
+	}
+	if len(logs) == 0 {
+		return &v1alpha2.ObservationLog{}, nil
+	}
+	olog, err := d.parseLogs(tId, strings.Split(string(logs), "\n"), objectiveValueName, metrics)
+	return olog, err
+}
+
+func (d *MetricsCollector) parseLogs(tId string, logs []string, objectiveValueName string, metrics []string) (*v1alpha2.ObservationLog, error) {
+	var lasterr error
+	olog := &v1alpha2.ObservationLog{}
+	mlogs := []*v1alpha2.MetricLog{}
+	for _, logline := range logs {
+		if logline == "" {
+			continue
+		}
+		ls := strings.SplitN(logline, " ", 2)
+		if len(ls) != 2 {
+			log.Printf("Error parsing log: %s", logline)
+			lasterr = errors.New("Error parsing log")
+			continue
+		}
+		_, err := time.Parse(time.RFC3339Nano, ls[0])
+		if err != nil {
+			log.Printf("Error parsing time %s: %v", ls[0], err)
+			lasterr = err
+			continue
+		}
+		kvpairs := strings.Fields(ls[1])
+		for _, kv := range kvpairs {
+			v := strings.Split(kv, "=")
+			if len(v) > 2 {
+				log.Printf("Ignoring trailing garbage: %s", kv)
+			}
+			if len(v) == 1 {
+				continue
+			}
+			metricName := ""
+			if v[0] == objectiveValueName {
```

Comment:
Earlier comment: Why do we need to distinguish between objectiveMetric and other additional  Metrics?